qtranslate_core.php

```diff
@@ -621,14 +621,27 @@ function qtranxf_load_option_flag_location( $nm ) {
     }
 }
 
-function qtranxf_validateBool( $var, $default_value ) {
-    if ( $var === '0' ) {
-        return false;
-    } elseif ( $var === '1' ) {
-        return true;
-    } else {
-        return $default_value;
+function qtranxf_validateBool( $var ) {
+    if (is_string( $var )){
+        $val = strtolower( $var );
+        switch ( $val ) {
+            case '0':
+            case 'n':
+            case 'no':
+                return false;
+            case '1':
+            case 'y':
+            case 'yes':
+                return true;
+        }
+    } elseif (is_array( $var )){
+        $result=array();
+        foreach ($var as $key => $element){
+            $result[$key]=qtranxf_validateBool($element);
+        }
+        return $result;
```

Comment:
I prefer not to mix basic/fundamental types with arrays here. So I'd abandon this part.